File: prepare/s1/src/w1/p07/Solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur_max = nums[0]
        glob_max = cur_max
        for i in range(1, len(nums)):
            cur_max = max(cur_max, 0) + nums[i]
            glob_max = max(glob_max, cur_max)
        return glob_max
        

File: prepare/s1/src/w1/p07/Solution.java
package w1.p07;
class Solution {
    
    public int maxSubArray(int[] nums) {
        
        int curMax = Integer.MIN_VALUE;
        int globMax = curMax;
        
        for(int i=0; i<nums.length; i++){
            curMax = Math.max(curMax,0) + nums[i];
            globMax = Math.max(globMax, curMax);
        }
        
        return globMax;
    }
}